internal/db/db.go

```diff
@@ -1,35 +1,71 @@
 package db
 
 import (
-	"database/sql"
 	"os"
+	"strconv"
 	"time"
 
-	_ "github.com/go-sql-driver/mysql"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
 )
 
 type Database interface {
 	Connect() error
 	RunMigrations()
-	Close() error
+	Find(interface{}, ...interface{}) error
+	Save(interface{}) error
+}
+
+func (db *PostgresDB) Find(data interface{}, conds ...interface{}) error {
+	db.tx = db.DB.Find(data, conds)
+	return db.tx.Error
 }
 
 type PostgresDB struct {
-	*sql.DB
+	*gorm.DB
```

Comment:
I would not expect to. An ORM should be the better way to go (I know Go devs don't typically like dependencies, but writing your own SQL is not a good idea unless you really understand how databases work and have AND use proper indexes)

Reply:
I heard someone else complain about Gorm recently. Is there another ORM that we should use? I'm generally opposed to not using an ORM as I'm of the personal opinion that too often developers think they should use SQL because they are developers but really they shouldn't because using SQL is more than just syntax, you need to understand the specific database internals and structure of the tables to write proper SQL. An ORM "saves" you from this as ORMs are typically built to manage the structure of the database with the understanding of the specific database (MySQL / Postgres / etc...) you are working with.  (also, ORM makes it easier to swap out to a different DB in the future as you avoid using syntax that is specific to one database)

Reply:
I've heard very good things about Ent. That being said I'm pretty sure SQLx works with any relational datastore and allows for schema-first querying.